doc/dev/debugging.rst

```diff
@@ -0,0 +1,310 @@
+        [RUNNING] quantum-opt --lower-gradients --convert-arraylist-to-memref /tmp/tmpwsoh3acq/circuit.nohlo.mlir -o /tmp/tmpwsoh3acq/circuit.nohlo.opt.mlir
+        [RUNNING] quantum-opt --inline --gradient-bufferize --scf-bufferize --convert-tensor-to-linalg --convert-elementwise-to-linalg --arith-bufferize --empty-tensor-to-alloc-tensor --bufferization-bufferize --tensor-bufferize --linalg-bufferize --tensor-bufferize --quantum-bufferize --func-bufferize --finalizing-bufferize --buffer-loop-hoisting --convert-bufferization-to-memref --canonicalize --cp-global-memref /tmp/tmpwsoh3acq/circuit.nohlo.opt.mlir -o /tmp/tmpwsoh3acq/circuit.nohlo.opt.buff.mlir
+        [RUNNING] quantum-opt --convert-linalg-to-loops --convert-scf-to-cf --expand-strided-metadata --lower-affine --arith-expand --convert-complex-to-standard --convert-complex-to-llvm --convert-math-to-llvm --convert-math-to-libm --convert-arith-to-llvm --finalize-memref-to-llvm=use-generic-functions --convert-index-to-llvm --convert-gradient-to-llvm --convert-quantum-to-llvm --emit-catalyst-py-interface --canonicalize --reconcile-unrealized-casts /tmp/tmpwsoh3acq/circuit.nohlo.opt.buff.mlir -o /tmp/tmpwsoh3acq/circuit.nohlo.opt.buff.llvm.mlir
+        [RUNNING] mlir-translate --mlir-to-llvmir /tmp/tmpwsoh3acq/circuit.nohlo.opt.buff.llvm.mlir -o /tmp/tmpwsoh3acq/circuit.nohlo.opt.buff.llvm.ll
+        [RUNNING] llc --filetype=obj --relocation-model=pic /tmp/tmpwsoh3acq/circuit.nohlo.opt.buff.llvm.ll -o /tmp/tmpwsoh3acq/circuit.nohlo.opt.buff.llvm.o
+        [RUNNING] clang -shared -rdynamic -Wl,-no-as-needed -Wl,-rpath,runtime/build/lib/capi:runtime/build/lib/backend:mlir/llvm-project/build/lib -Lmlir/llvm-project/build/lib -Lruntime/build/lib/capi -Lruntime/build/lib/backend -lrt_backend -lrt_capi -lpthread -lmlir_c_runner_utils /tmp/tmpwsoh3acq/circuit.nohlo.opt.buff.llvm.o -o /tmp/tmpwsoh3acq/circuit.nohlo.opt.buff.llvm.so
+
+
+Pass Pipelines
+==============
+
+The compilation steps which take MLIR as an input and lower it to binary are broken into pass pipelines.
+A ``PassPipeline`` is a class that specifies which binary and which flags are used for compilation.
+Users can implement their own ``PassPipeline`` by inheriting from this class and implementing the relevant methods/attributes.
+Catalyst's compilation strategy can then be adjusted by overriding the default pass pipeline.
+For example, let's imagine that a user is interested in testing different optimization levels when compiling LLVM IR to binary using ``llc``.
+The user would then create a ``PassPipeline`` that replaces the ``LLVMIRToObjectFile`` class.
+First let's take a look at the ``LLVMIRToObjectFile``.
+
+.. code-block:: python
+    class LLVMDialectToLLVMIR(PassPipeline):
+        """LLVMIR To Object File."""
+    
```

Comment:
It doesn't show the code-block in the compiled docs.
Update this as follow to fix the issue:
```suggestion
.. code-block:: python

    class LLVMIRToObjectFile(PassPipeline):
        """LLVMIR To Object File."""
    
```